torch/_inductor/fx_passes/quantization.py

```diff
@@ -2529,6 +2529,190 @@ def _register_qlinear_weight_prepack():
         )
 
 
+def _register_smooth_quant_int_mm_pattern():
+    """
+    The pattern is:
+      (no bias) reshape -> _int_mm -> convert_element_type -> (expand -> mul) -> mul -> reshape
+    or
+      (with bias) pattern_no_bias -> add -> reshape -> reshape
+    """
+    pattern_no_bias = CallFunction(
+        aten.reshape.default,
+        CallFunction(
+            aten.mul.Tensor,
+            CallFunction(
+                aten.mul.Tensor,
+                CallFunction(
+                    prims.convert_element_type.default,
+                    CallFunction(
+                        aten._int_mm.default,
+                        CallFunction(
+                            aten.reshape.default,
+                            KeywordArg("a"),
+                            KeywordArg("in_shape"),
+                        ),
+                        KeywordArg("b"),
+                    ),
+                    KeywordArg("dtype"),
+                ),
+                CallFunction(
+                    aten.expand.default,
+                    KeywordArg("x_scale"),
+                    Arg(),
+                ),
+            ),
+            KeywordArg("w_scale"),
+        ),
+        KeywordArg("out_shape_no_bias"),
+    )
+    pattern_with_bias = CallFunction(
+        aten.reshape.default,
+        CallFunction(
+            aten.reshape.default,
+            CallFunction(
+                aten.add.Tensor,
+                pattern_no_bias,
+                KeywordArg("bias"),
+            ),
+            Arg(),
+        ),
+        KeywordArg("out_shape_with_bias"),
+    )
+
+    def _validate_pattern(match: Match):
+        for n in match.nodes:
+            if len(n.users) != 1 and n != match.output_node():
```

Comment:
I think it's not needed of `len(n.users) != 1`. The pattern matcher should cover this check.

Reply:
Thanks. Removed.